kinto/core/permission/memory.py

```diff
@@ -103,7 +103,7 @@ def get_accessible_objects(self, principals, bound_permissions=None, with_childr
         else:
             for pattern, perm in bound_permissions:
                 id_match = ".*" if with_children else "[^/]+"
-                regexp = re.compile(f"^{pattern.replace('*', id_match)}$")
+                regexp = re.compile(f"^{re.escape(pattern).replace('*', id_match)}$")
```

Comment:
Good call! But I suspect that this is also escaping patterns that we use internally to match multiple objects.

I think this could be done around here instead maybe: https://github.com/Kinto/kinto/blob/f27fb5ae974fabcb6618ee2026ad748b8807fae4/kinto/core/authorization.py#L236-L261